### ISSUE TITLE
potential memory leak issues

### ISSUE THREAD

**Issue by xinyuwufei** [qixiang109/seq2seq-autoencoder#1]
HI, qixiang109
first , thanks for very clear and neat code of implementation.

I found a small memory leak issue when  running large dataset
refer to this:
http://stackoverflow.com/documentation/tensorflow/3883/how-to-debug-a-memory-leak-in-tensorflow/13426/use-graph-finalize-to-catch-nodes-being-added-to-the-graph#t=201702020243343079787

the program continually eats the memory and eventually is killed, you can verify this by something like
'ps au | grep python3 ...' to check the memory usage. 

issue is caused by this line:
encoder_states,decoder_outputs,loss,_ = session.run([self.encoder_states,tf.transpose(tf.pack(self.decoder_outputs),[1,0,2]),self.loss,self.update],feed)
and this line:
encoder_states,decoder_outputs,loss = session.run([self.encoder_states,tf.transpose(tf.pack(self.decoder_outputs),[1,0,2]),self.loss],feed)

we should not add/assign tf op insde the seesion.run arguments

 :‑)
